katas/content/complex_arithmetic/polar_multiplication/Solution.qs

```diff
@@ -3,10 +3,10 @@ namespace Kata {
 
     function ComplexPolarMult(x : ComplexPolar, y : ComplexPolar) : ComplexPolar {
         mutable theta = x.Argument + y.Argument;
-        if theta > PI() {
+        while theta > PI() {
```

Comment:
Nice catch. I think that this task assumes the given complex numbers x and y have −π < θ ≤ π , which is the principal argument. That's why if condition is used. Although, QDK in itself doesnt throw an error if one provides  argument θ, which doesnt follow these bounds.  That being said there is definitely room for improvement. 

Quoting verbation about "principal argument" from ["5.6 Argument of a complex number"](http://www.numbertheory.org/book/cha5.pdf) from [Elementary Linear Algebra by [Keith Matthews](http://www.numbertheory.org/keith.html)](http://www.numbertheory.org/book/)
> "Let z = x + iy be a non–zero complex number, r = |z| = \sqrt{x^2+y^2}. Then
we have x = r cos θ, y = r sin θ, where θ is the angle made by z with the
positive x–axis. So θ is unique up to addition of a multiple of 2π radians.
Any number θ satisfying the above pair of equations is called an argument of z and is denoted by arg z. The **particular argument of z** lying in the range −π < θ ≤ π is called the principal
argument of z and is denoted by Arg z (see Figure 5.5)." 

### Approach 1
I would suggest to improve the "Polar Cordinates" lesson to include this note about the principal argument of a complex number and subsequent task descriptions to include this information. For example, "Polar multiplication" task could be rephrased as :

1. A complex polar number $x = r_{1}e^{i\theta_1}$ in its principal form, i.e. $r_1 \geq 0$ and $-\pi < \theta_1 \leq \pi$.
2. A complex polar number $y = r_{2}e^{i\theta_2}$  in its principal form, i.e. $r_2 \geq 0$ and $-\pi < \theta_2 \leq \pi$.. 

 ### Approach 2
 Keep this solution. Change testing strategy to explicitly generate the magnitude and theta. Then, update testing strategy to fail the complex numbers where someone has used `if` condition instead of `while` loop.

Reply:
I was also thinking about edge case testing but more from a brute force angle to find if there was one. I never quite got the hang of trigonometry and spent way too long trying to somewhat understand this lesson/solution. Really appreciate you including links where the problems are discussed in more detail!

Reply:
Theoretically, if you see the addition/subtraction of $2*\pi$ doesnt make difference to the polar complex number because $e^{\iota.(2n\pi+\theta)} = cos(2n\pi+\theta) + \iota . sin(2n\pi+\theta) = cos(\theta) + \iota . sin(\theta) = e^{\iota\theta}$. 

Its just the thing that for a complex number $z = r e^{\iota \theta}$,  the principal form suggests that $r \geq 0$ and $-\pi < \theta \leq \pi$. Think of it as a convention. For example: $z = 2 . e^{\iota \pi /4} =  2 . e^{\iota (\pi /4 + 2 \pi)} = 2 . e^{\iota (\pi /4 + \pi)} . e^{\iota (\pi)} = -2 . e^{\iota (\pi /4 + \pi)}$, but out of these variants we pick $z = 2 . e^{\iota \pi /4}$, since $2>=0$ and $-\pi < \pi/4 \leq \pi$. [($e^{\iota (\pi)}=-1$) , since you are wondering about (-)ve sign]

Reply:
I tend to agree with the first approach @Manvi-Agrawal suggested, where the fix will be in text for the lesson to clarify the inputs are expected within a specific range. While this solution is more generally correct, I don't think updating the tests to fail user solutions that don't accommodate values outside the range is that helpful for what is meant to be an introductory task.

@cesarzc, @tcNickolas any preference here?

Reply:
I also agree with the first approach. Changing test should be done carefully since users' solutions persist in the [Azure Quantum katas environment](https://quantum.microsoft.com/en-us/tools/quantum-katas). If users start getting error from previously accepted solutions, that results in a bad experience.

Reply:
I also agree the with approach 1 being a better solution, my confusion was due to not understanding the concepts behind the problem.

Reply:
@HopeAnnihilator, I was wondering if you would have some time to do the changes to this PR to align with the agreed approach? Is there anything else we can help you with?

Reply:
@HopeAnnihilator do you mean markdown formatting guidelines? If so, we don't have a specific set of guidelines to format markdown, the main thing to look for is that the kata content renders as expected.

This [readme](https://github.com/microsoft/qsharp/blob/main/katas/README.md) file contains the information needed to build the katas locally and verify they render correctly.

Reply:
I was more concerned with phrasing and such but I doubt it will be too much of an issue. I run everything locally and will look to get on it tomorrow. Life been a bit hectic these past few weeks but I should be able to throw together something.

Reply:
Thank you for following up on this @HopeAnnihilator!

Reply:
I am not experienced in math speak but I believe that should be a bit more clear @cesarzc